### PR TITLE
feat: support IP allow list

### LINKED ISSUE #680: feature: support IP restriction
relate to  #620

- [ ] support IP allow list

### PR DESCRIPTION
**Pull request by nic-chen**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues
close #680

___
### New feature or improvement
- Describe the details and related test reports.

support IP allow list for Manager API

### CODE REVIEW

File: api/internal/conf/conf.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ var (
 	ENV              string
 	Schema           gjson.Result
 	WorkDir          = "."
-	ServerHost       = "127.0.0.1"
+	ServerHost       = ""
```
Comment (starsz) [inline api/internal/conf/conf.go:46]: IMO, I think we can still keep this. For someone who only wants to bind a specific IP.

File: api/internal/filter/ip_filter.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+			return
+		}
+
+		if ipStr == "" {
+			log.Warn("forbidden by empty IP")
+			c.AbortWithStatusJSON(http.StatusForbidden, consts.ErrIPNotAllow)
+			return
+		}
+
+		ips, subnets := generateIPSet(conf.AllowList)
```
Comment (starsz) [inline api/internal/filter/ip_filter.go:96]: There is no need to generateIPSet for every request.

**Review by starsz — COMMENTED** [review]
Can we add cli test for this PR?

File: api/internal/conf/conf.go (hunk 1 above)
Comment (nic-chen) [reply api/internal/conf/conf.go]: fixed.

File: api/internal/filter/ip_filter.go (hunk 2 above)
Comment (nic-chen) [reply api/internal/filter/ip_filter.go]: fixed.

**Review by starsz — APPROVED** [review]

**Comment by imjoey** [comment]
@nic-chen This is a great and useful feature for users. How about add relevant docs in user guide? Thanks.

**Comment by imjoey** [comment]
> @nic-chen This is a great and useful feature for users. How about add relevant docs in user guide? Thanks.

we could merge this first and then add docs in another or. what's you opinions? @nic-chen @juzhiyuan @starsz

**Review by imjoey — APPROVED** [review]
/lgtm

**Comment by starsz** [comment]
> > @nic-chen This is a great and useful feature for users. How about add relevant docs in user guide? Thanks.
> 
> we could merge this first and then add docs in another PR. what's you opinions? @nic-chen @juzhiyuan @starsz

Agree with it.
Maybe we can support the blacklist. 😄

**Comment by nic-chen** [comment]
> > @nic-chen This is a great and useful feature for users. How about add relevant docs in user guide? Thanks.
> 
> we could merge this first and then add docs in another PR. what's you opinions? @nic-chen @juzhiyuan @starsz

Thanks, I will create an issue to track it.

**Comment by juzhiyuan** [comment]
ok

**Pull request merged**